crates/language/src/language_registry.rs

```diff
@@ -731,20 +737,37 @@ impl LanguageRegistry {
         });
         self.find_matching_language(move |language_name, config, current_best_match| {
             let path_matches_default_suffix = || {
-                config
-                    .path_suffixes
-                    .iter()
-                    .any(|suffix| path_suffixes.contains(&Some(suffix.as_str())))
+                let len =
+                    config
+                        .path_suffixes
+                        .iter()
+                        .fold(0, |acc: usize, path_suffix: &String| {
+                            let ext = ".".to_string() + path_suffix;
+
+                            let matched_suffix_len = path_suffixes
+                                .iter()
+                                .find(|(suffix, _)| suffix.ends_with(&ext) || suffix == path_suffix)
```

Comment:
This latest version is implicitly depending on the above defined `path_suffixes` to be sorted by length, so that find will find return the shortest matching element, and prevent us having to look at every element. I may add a `debug_assert!()` for that.